### ISSUE TITLE
When "dotnet restore" fails, gulp task is not failing

### ISSUE THREAD

**Issue by IgorYastrebov** [Janus-vistaprint/gulp-dotnet-cli#27]
configure your .net core app with a wrong nuget packages
gulp.src(...)
.pipe(restore(...))
.pipe(msbuild(...))

the problem is that .pipe(restore()) is not erroring a pipe.

**Comment by TerribleDev** [comment]
You are half correct.

I've done some digging here.

We do return to gulp a promise, which will signal to gulp that an error has occured.


for the following example if you run gulp build, build will never fire because the restore task fails. 

```js
gulp.task('restore', ()=>{
    return gulp.src('**/*.csproj')
        .pipe(restore())
});
gulp.task('build', ['restore'], ()=>{
    return gulp.src('**/*.csproj')
        .pipe(build());
});
```

You are correct that in your example if restore fails build will still fire. However if you `return gulp.src().pipe(..)` gulp will know something has failed in the task. That being said this is less than ideal, because the second pipe should never trigger if the first errored. The problem is that we don't signal to the vinyl stream that there has been an issue during each pipe. I'm going to have to look into how to handle this case correctly.

**Comment by TerribleDev** [comment]
Quick update,

@IgorYastrebov I believe we might handle the correct behavior as far as gulp is concerned as a plugin. Here is some documentation: https://gist.github.com/floatdrop/8269868

you can use gulp plumber to short circuit out of errors prematurely: https://github.com/floatdrop/gulp-plumber

**Comment by TerribleDev** [comment]
ok half way, it seems we still might not emit the correct error.

**Comment by TerribleDev** [comment]
So I have found out when we have an error we still pass along the file in the stream. We should only return the error to vinyl to short circuit this.

**Comment by TerribleDev** [comment]
We emitted the error in the callback, but not directly in the stream.

**Comment by TerribleDev** [comment]
Fixed in 1.1.0
